### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Richaa2**

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        Random random = new Random();
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: lets make Random as class level variable, not to create new instance of Random class every time we call getRandomColor() method

File: src/main/java/core/basesyntax/Colors.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Colors.java:3]:
```suggestion
public enum Color {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        Random random = new Random();
+        Ball ball = new Ball();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(100));
+        return ball;
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/Lottery.java:12]: same, let's make Random and ColorSupplier as class level variables, not to create new instances every time we call getRandomBall() method. And lets make number 100 as constant field https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Review by Andr1yK — CHANGES_REQUESTED** [review]
test

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 4]
```diff
@@ -1,7 +1,13 @@
...
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private Random random = new Random();
+
     public String getRandomColor() {
-        return null;
+        final int lengthOfColors = Color.values().length;
+        Color values = Color.values()[random.nextInt(lengthOfColors)];
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorSupplier.java:10]:
```suggestion
        Color color = Color.values()[random.nextInt(lengthOfColors)];
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+public class Lottery {
+    public static final int NUMBER = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Ball getRandomBall() {
+        Ball ball = new Ball();
+        ball.setColor(colorSupplier.getRandomColor());
+        ball.setNumber(random.nextInt(NUMBER));
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Lottery.java:13]:
```suggestion
        ball.setNumber(random.nextInt(MAX_NUMBER));
```

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**